src/v2/Apps/Artist/Components/ArtistTopWorksRail/ArtistTopWorksRail.tsx

```diff
@@ -0,0 +1,105 @@
+import { Box, Flex, Sans } from "@artsy/palette"
+import React from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { ArtistTopWorksRail_topRailArtworks } from "v2/__generated__/ArtistTopWorksRail_topRailArtworks.graphql"
+import { Carousel } from "v2/Components/Carousel"
+import FillwidthItem from "v2/Components/Artwork/FillwidthItem"
+import { Media } from "v2/Utils/Responsive"
+import { StyledLink } from "v2/Apps/Artist/Components/StyledLink"
+
+interface ArtistTopWorksRailProps {
+  topRailArtworks: ArtistTopWorksRail_topRailArtworks
+}
+
+export const ArtistTopWorksRail: React.FC<ArtistTopWorksRailProps> = ({
+  topRailArtworks,
+}) => {
+  const carouselHeight = 300
+  const artworks = topRailArtworks?.images?.edges ?? []
+  const handleViewWorksClick = () => {
+    // no-op
+  }
+  return artworks.length > 0 ? (
```

Comment:
```
if (artworks.length === 0) {
  return null
}

return (...) 
```

Reply:
Reading this over, and while the logic is clear: "for each of mobile/desktop, render the rail, as well as a conditional handling of the 'view all' link based on whether this is the overview tab or not", I wonder if that can be made more readable via something like:

``` ts
const Rail = <Carousel ... />
const ViewAllLink = onOverViewTab ? ... : ...
const Title = ... 

return (
  <Media at="xs">
    <Title />
    <VewAll />
    <Carousel />
)
```

kind of thing.

(Sidenote, it doesn't look like the `xs` and desktop layout requires anything different? they _seem_ the same to me. I think one might say 'view all', and the other 'view all artworks', but I think there's an easy way to do that in one line vs using the `Media` helper).